src/engine/commands.ts

```diff
@@ -101,6 +104,30 @@ export const nip98Fetch = async (url, method, body = null) => {
   return Fetch.fetchJson(url, {body, method, headers})
 }
 
+export const makeDvmRequest = (request: DVMRequestOptions & {delay?: number}) => {
+  const emitter = new Emitter()
+  const {event, relays, timeout = 30_000, autoClose = true, reportProgress = true} = request
+  const kind = event.kind + 1000
+  const kinds = reportProgress ? [kind, 7000] : [kind]
+  const filters: Filter[] = [{kinds, since: now() - 60, "#e": [event.id]}]
+
+  const sub = subscribe({relays, timeout, filters})
+  const thunk = publish({event, relays, timeout, delay: request.delay})
+
+  sub.emitter.on(SubscriptionEvent.Event, (url: string, event: TrustedEvent) => {
+    if (event.kind === 7000) {
+      emitter.emit(DVMEvent.Progress, url, event)
+    } else {
+      emitter.emit(DVMEvent.Result, url, event)
+
+      if (autoClose) {
+        sub.close()
+      }
+    }
+  })
+  return {request, emitter, sub, thunk}
+}
+
```

Comment:
Why not use `@welshman/dvm`?

Reply:
Just to get a Thunk with the delay, the toaster also understand thunk better to display the request status

Reply:
Gotcha, I guess that's ok